modules/farm_fd2/src/entrypoints/cover_crop_seeding/lib.js

```diff
@@ -0,0 +1,292 @@
+          );
+        }
+      },
+    };
+    ops.push(areaSeededQuantity);
+
+    if (formData.winterKill) {
+      const winterKillLog = {
+        name: 'winterKillLog',
+        do: async (results) => {
+          return await farmosUtil.createWinterKillActivityLog(
+            formData.winterKillDate,
+            formData.location,
+            formData.beds,
+            results.plantAsset
+          );
+        },
+        undo: async (results) => {
+          await farmosUtil.deleteWinterKillActivityLog(
+            results['winterKillLog'].id
+          );
+        },
+      };
+      ops.push(winterKillLog);
+    }
+
+    const seedingLog = {
+      name: 'seedingLog',
+      do: async (results) => {
+        return await farmosUtil.createSeedingLog(
+          formData.date,
+          formData.location,
+          formData.beds,
+          ['seeding', 'seeding_cover_crop'],
+          results.plantAsset,
+          [results.areaSeededQuantity]
+        );
+      },
+      undo: async (results) => {
+        await farmosUtil.deleteSeedingLog(results['seedingLog'].id);
+      },
+    };
+    ops.push(seedingLog);
+
+    if (formData.seedApplicationEquipment.length > 0) {
+      const seedApplicationDepthQuantity = {
+        name: 'seedApplicationDepthQuantity',
+        do: async () => {
+          return await farmosUtil.createStandardQuantity(
+            'length',
+            formData.seedApplicationDepth,
+            'Depth',
+            'INCHES'
+          );
+        },
+        undo: async (results) => {
+          await farmosUtil.deleteStandardQuantity(
+            results['seedApplicationDepthQuantity'].id
+          );
+        },
+      };
+      ops.push(seedApplicationDepthQuantity);
+
+      const seedApplicationSpeedQuantity = {
+        name: 'seedApplicationSpeedQuantity',
+        do: async () => {
+          return await farmosUtil.createStandardQuantity(
+            'rate',
+            formData.seedApplicationSpeed,
+            'Speed',
+            'MPH'
+          );
+        },
+        undo: async (results) => {
+          await farmosUtil.deleteStandardQuantity(
+            results['seedApplicationSpeedQuantity'].id
+          );
+        },
+      };
+      ops.push(seedApplicationSpeedQuantity);
+
+      for (const equipmentName of formData.seedApplicationEquipment) {
+        seedApplicationEquipmentAssets.push(equipmentMap.get(equipmentName));
+      }
+
+      const seedApplicationActivityLog = {
+        name: 'seedApplicationActivityLog',
+        do: async (results) => {
+          return await farmosUtil.createSoilDisturbanceActivityLog(
+            formData.date,
+            formData.location,
+            formData.beds,
+            ['tillage', 'seeding_cover_crop'],
+            results.plantAsset,
+            [
+              results.seedApplicationDepthQuantity,
+              results.seedApplicationSpeedQuantity,
+              results.areaSeededQuantity,
```

Comment:
Unfortunately, reusing the `areaSeededQuantity` creates problems.  When a log is deleted, all of the quantities that it references are also deleted.  So here, if the seeding log were deleted, then this log would refer to a quantity that no longer exists.  So best to create a new quantity for the area seeded for each of the logs. Also, add to the tests to check that each of the new quantities is associated with the correct log.